hw/top_earlgrey/data/ip/chip_entropy_src_testplan.hjson

```diff
@@ -179,5 +179,74 @@
       tests: []
       bazel: []
     },
+    {
+      name: chip_sw_entropy_src_fw_observe_many_contiguous
+      desc: '''Verify that the software can observe samples fast enough to extract many contiguous samples.
+
+            Procedure:
+            - Disable the entropy complex.
+            - Configure realistic health test threshold values for FIPS/CC compliant mode.
+            - Enable ENTROPY_SRC in FIPS mode (FIXME: is it necessary to use FIPS mode?).
+            - Enable Firmware Override - Observe.
+            - Collect 1000 samples from the observe FIFO:
+              - Verify after reading each sample that the observe FIFO overflow status has not been set.
+              - FIXME: should the SW run a health check to make sure that it is fast enough?
+
+            Tests:
+            - The procedure must be run once with ENTROPY_SRC.RNG_BIT_ENABLE disabled (4-bit / sample) and
+              four times with the ENTROPY_SRC.RNG_BIT_ENABLE feature being enabled (once for every channel).
+
+            Notes for silicon targets:
+            - This test requires access to the ENTROPY_SRC.FW_OV.OBSERVE feature, which is gated by OTP.
+              This test might thus need to be done in a secure facility.
+            '''
+      features: [
+        "ENTROPY_SRC.MODE.FIPS", // FIXME: maybe
+        "ENTROPY_SRC.FW_OV.OBSERVE"
+        "ENTROPY_SRC.RNG_BIT_ENABLE",
+      ]
+      stage: V2
+      si_stage: SV3
+      lc_states: ["TEST_UNLOCKED", "PROD"]
+      tests: []
+      bazel: []
+    },
+    {
+      name: chip_sw_entropy_src_fw_extract_and_insert
+      desc: '''Verify that the software can observe samples fast enough to extract many contiguous samples.
+
+            Procedure:
+            - Disable the entropy complex.
+            - Configure realistic health test threshold values for FIPS/CC compliant mode.
+            - Enable ENTROPY_SRC in FIPS mode (FIXME: is it necessary to use FIPS mode?)
+            - Enable Firmware Override - Extract & Insert.
+            - Enable CSRNG.
+            - Enable both EDNs in auto request mode.
+            - Trigger the execution of various cryptographic hardware blocks consuming entropy to stress test the entropy complex.
```

Comment:
Here you could use the power_virus test as a reference. It would also be nice if you could verify that SW can receive entropy.

Reply:
Just a quick question since it wasn't entirely clear to me from the documentation: for the SW to receive entropy, does it need to disable the HW flow? ie https://opentitan.org/book/hw/ip/entropy_src/doc/programmers_guide.html#reading-entropy-output
Or are you referring to another mecanism?

Reply:
There are two different ways to extract entropy:
- The observe FIFO: gives entropy from before the conditioner
- The ENTROPY_DATA register: gives the final entropy output by the entropy_src, can be with or without conditioning.

The Firmware Override modes give access to the Observe FIFO only.